workshops/templatetags/navigation.py

```diff
@@ -12,23 +34,38 @@ def navbar_element(context, title, url_name):
     is later reversed into proper URL.
     """
     url = reverse(url_name)
-    return navbar_element_url(context, title, url)
+    active = context['request'].path == url
+    return make_element(title, url, active=active)
 
 
 @register.simple_tag(takes_context=True)
-def navbar_element_url(context, title, url):
+def navbar_element_permed(context, title, url_name, perms):
     """
-    Insert Bootstrap's `<li><a>...</a></li>` with specific classes and
-    accessibility elements.  This tag takes a pre-made URL as an argument.
+    Works like `navbar_element`, but also disables if user doesn't have
+    permissions.
+
+    `perms` can be a comma-separated string of permissions or a single
+    permission.
     """
-    request = context['request']
+    url = reverse(url_name)
+    active = context['request'].path == url
+    disabled = True
 
-    active = ""
-    screen_reader = ""
+    # check permissions
+    perms_ctx = context['perms']
+    perms = perms.split(',')
+    perms = map(lambda x: x in perms_ctx, perms)
+    if all(perms):
+        disabled = False
```

Comment:
Why not just `disabled = not all(perms)` and get rid of the default setting to `True` above?